WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java

```diff
@@ -751,7 +751,7 @@ private static String getSortColumnForTag(ReaderTag tag) {
         } else if (tag.isTagTopic() || tag.isBookmarked()) {
             return "date_tagged";
         } else {
-            return "date_published";
+            return "datetime(date_published)";
```

Comment:
❤️ So it seems like the issue is that we store this date as a string but it has format like "2024-01-08T14:42:19+00:00", and not all posts have the same timezone, causing the order to be inconsistent since the backend actually treats the field as a date. This is a really nice catch, thanks!

💡  Also, since this is the issue, I think it makes sense to add `datetime` around the other date-related cases above (lines 746, 748, and 752), since they would suffer the same problem.

💡 Do you think it also makes sense to use the `datetime` function on other queries in this file that use a date as `order` clause? ([here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L661), [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L668), [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L985), [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L1023), and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L1100))

Reply:
Thanks for the review, @thomashorta !

> 💡 Also, since this is the issue, I think it makes sense to add datetime around the other date-related cases above (lines 746, 748, and 752), since they would suffer the same problem.

>💡 Do you think it also makes sense to use the datetime function on other queries in this file that use a date as order clause? ([here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L661), [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L668), [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L985), [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L1023), and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/abc64da44acaa7df5b4f02fa69815d7f93e6bc9e/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L1100))

I agree 100%, but this also means we'd have to test every single scenario that relies on the logic you've mentiond in this same PR, right? Since this already solves the original tag post list issue, does it make sense to create a separate issue for the other places where we can thoroughly investigate and test the new changes?

Reply:
That makes sense, let's make the `datetime` change for the other places in a different PR then.

Reply:
[Issue created](https://github.com/wordpress-mobile/WordPress-Android/issues/19897).